### DISCUSSION TITLE
Hi Everyone!

### DISCUSSION THREAD

**Discussion by Pand4717** [evennia/evennia#2400] [Greetings & Introductions]
My name is Jay and I've played MUSHes since 1998. The same one in fact, with occasional others. Over the past year I've had two small strokes (my typing is starting to come back but not nearly as good as before), my kidneys failed and I'm on dialysis, and I had a heart attack and a quadruple bypass for my 40th birthday this past march Now I'm disabled. How was your COVID year, heh!

I'm telling you this because I wanted to say that the universe tried to kill me many times this year, and I'm still here. I'm not working and I'm recuperating, and having lived I've decided that I ought to get to accomplishing my dreams. One that I can definitely work on right now is the game I've always wanted to program.

I'm familiar with how MUSH soft code works, but alas beyond simple HTML that is the extent  of my coding experience. I've looked at various formats: Pennmush, AresMUSH, and others. I finally settled on Evennia. I've spent the last week pouring over the documentation and code, and I think I'm starting to get it. Evennia is so well documented I think I actually move forward with my dream.

WARNING! DREAM FOLLOWS. IT'S LONG.

My dream game is a cross canon MUSH/Mud hybrid based on the Dark Tower series. My dream includes a complex environment comprised of universes, or levels of the tower, stacked right on top of each other. It will include both a turn-based space system based on TurnBattle and a twitch combat system incorporating the Traits system.

The game will be designed as a game for writers: my goal is to offer tools to scratch that grinding combat itch, while incorporating and incentivizing  narrative objectives within the coded systems themselves. Examples include combat poses as a coded system, assigned a score and added as a bonus or buff - or what about incentivizing IC journal keeping? And offering players the opportunity nominate a journal of the week?  

The other would be an @quest system, which will send players across their level of the tower and maybe even to another, stopping in different rooms, interacting with spawned npcs and usually fighting a battle or two. Everything the players say, emote, and fire in terms of battle poses will be counted up and given a score (could i use something like Readability https://pypi.org/project/readability/?) The 'RP Score' will be critical to quest payout.

There's lots more but I think I'll just leave it there for now. I've been playing text based games for nearly 30 years and I've been percolating these ideas for that entire time. Any feedback you have about what I've said so far would be extremely appreciated. You'll probably see me around, I have lots of questions.

Be well!

-Jay

“Yet suppose further. Suppose that all worlds, all universes, met at a single nexus, a single pylon, a Tower. And within it, a stairway, perhaps rising to the Godhead itself. Would you dare climb to the top, gunslinger? Could it be that somewhere above all of endless reality, there exists a room?...'

You dare not.'

And in the gunslinger's mind, those words echoed: You dare not.”
― Stephen King, The Gunslinger

**Comment by Griatch** [comment]
Welcome to Evennia, Jay!

A game based on the Dark Tower sounds interesting! The only advice I can give starting out is to start out _small_. When you start out (especially when you have little programming experience) it can be very hard to scope things and know what is hard to do and what is not. So I strongly suggest a minimal starting goal you can then expand upon later. Take a look at the [Game Planning](https://www.evennia.com/docs/latest/Game-Planning.html) page if you haven't already. 

If you have questions, don't be shy to ask in these forums, or in one of the online chat channels (discord or IRC, your choice). Good luck and welcome again!

**Reply by Pand4717** [reply]
I know it's one step at a time. I've been a nonprofit project manager for a while, and an educator, So I guess my hope is to build momentum, lean on some of my programming friends, and maybe see if I can commission some help. I know it's a big dream, but I guess I'm not up to a whole lot otherwise. One can only watch so many episodes of SVU.

**Reply by pruuk** [reply]
This is a fantastic idea for a MUD. I really love the Dark Tower Universe.

As another person with a dream and very little coding experience, my route was to just go for it on a small scale, then bin it, take some online courses, then restart, then bin it again, learn some more, then restart a third time. I'm on the third go on my MUD and it is starting to look like a real game now. I've reused some of the ideas from v1 and v2, but mostly started from scratch each time.
